server/src/main/java/org/elasticsearch/search/profile/AbstractInternalProfileTree.java

```diff
@@ -48,7 +48,7 @@ public AbstractInternalProfileTree() {
      * @param query The scoring query we wish to profile
      * @return      A ProfileBreakdown for this query
      */
-    public PB getProfileBreakdown(E query) {
```

Comment:
could you make all the fields `private` ? Otherwise inheriting classes can access them non-synchronized.

Reply:
thank you for catching that, it's important.